server/lib/src/server/access/mod.rs

```diff
@@ -318,33 +345,61 @@ pub trait AccessControlsTransaction<'a> {
         name = "access::search_filter_entry_attributes",
         skip_all
     )]
-    fn search_filter_entry_attributes(
-        &self,
+    fn search_filter_entry_attributes<'b>(
+        &'b self,
         se: &SearchEvent,
         entries: Vec<Arc<EntrySealedCommitted>>,
-    ) -> Result<Vec<Entry<EntryReduced, EntryCommitted>>, OperationError> {
+    ) -> Result<Vec<EntryReducedCommitted>, OperationError> {
+        struct DoEffectiveCheck<'b> {
+            modify_related_acp: Vec<AccessControlModifyResolved<'b>>,
+            delete_related_acp: Vec<AccessControlDeleteResolved<'b>>,
+            sync_agmts: &'b HashMap<Uuid, BTreeSet<Attribute>>,
+        }
+
+        let ident_uuid = match &se.ident.origin {
+            IdentType::Internal => {
+                // In production we can't risk leaking data here, so we return
+                // empty sets.
+                security_critical!("IMPOSSIBLE STATE: Internal search in external interface?! Returning empty for safety.");
+                // No need to check ACS
+                return Err(OperationError::InvalidState);
+            }
+            IdentType::Synch(_) => {
+                security_critical!("Blocking sync check");
+                return Err(OperationError::InvalidState);
+            }
+            IdentType::User(u) => u.entry.get_uuid(),
+        };
+
         // Build a reference set from the req_attrs. This is what we test against
         // to see if the attribute is something we currently want.
 
         // Get the relevant acps for this receiver.
-        let related_acp = self.search_related_acp(&se.ident);
-        let related_acp: Vec<_> = if let Some(r_attrs) = se.attrs.as_ref() {
-            // If the acp doesn't overlap with our requested attrs, there is no point in
-            // testing it!
-            related_acp
-                .into_iter()
-                .filter(|acs| !acs.acp.attrs.is_disjoint(r_attrs))
-                .collect()
-        } else {
-            related_acp
-        };
+        let search_related_acp = self.search_related_acp(&se.ident, se.attrs.as_ref());
```

Comment:
since this isn't used until line 402 wouldn't it make sense to move it under line 396? Because it currently gives the impression that it relates to the block in 380 which it does not.

Reply:
Technically it doesn't matter either way, since there is no data dependence, the compiler can move it above or below this, but I'll move it.